### PR TITLE
Add tests for tcp and epc modes

### PR DESCRIPTION
**Pull request by yuutayamada**
Added some basic tests for tcp mode.
Note that I couldn't pass a test for `use` command, so I just commented out.
Should I remove them?

### CODE REVIEW

**Comment by Araq** [comment]
> Added some basic tests for tcp mode.

Impressive. Love it. :-)

> Note that I couldn't pass a test for use command, so I just commented out.

Hmm, that's strange.

**Comment by yuutayamada** [comment]
>>    Note that I couldn't pass a test for use command, so I just commented out.

> Hmm, that's strange.

Oh, sorry. I should have written more explanation. There is nothing wrong with nimsuggest. Somehow my test code isn't working with only `use` command. (It seems nimsuggest returns right result), but I'm not sure how to fix.

**Comment by yuutayamada** [comment]
I cherry picked epc's tests from major-restructure branch, but tests didn't pass in this branch :(

File: tests/epc_suggest_test [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (Araq) [inline tests/epc_suggest_test]: The tests should also work on Windows. Write a proper Nim program that performs the testing please.

Also: Files should have file extensions.

File: tests/expected/epc_chk [hunk 2]
```diff
@@ -0,0 +1 @@
+(return 1 "Hint: sample [Processing]\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(4, 14) Error: invalid indentation\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(5, 3) Error: expression expected, but found 'Return hello'\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(4, 14) Error: 'string' cannot be assigned to\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(6, 3) Error: invalid indentation\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(6, 3) Error: value of type 'string' has to be discarded\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(13, 1) Error: undeclared identifier: 'he'\u000A<<<__TEST_DIRECTORY__>>>/sample.nim(4, 6) Error: implementation of 'sample.hello()' expected\u000A")
```
Comment (Araq) [inline tests/expected/epc_chk]: Files should have file extensions.

**Comment by Araq** [comment]
This way to do it is hard to maintain and requires a heap of different files to add a single new test case. Instead copy nim's testament design where the expected output is in a big ``discard """ ... """`` section at the top of the nim file. Also instead of giving column, line information explicitly, mark the "cursor"  via  ``#[]#`` that you remove from the nim file before giving it to nimsuggest.

**Comment by yuutayamada** [comment]
Sorry for late reply. I'll try another PR, but for some reasons (job hunting, may need to relocate, and nimsuggest's master and major-restructure branches no longer work with nim's latest version in my emacs), I think I'll take time. So go ahead making new test scheme. (I mean that tester thing)

**Pull request closed without merging**